src/coreclr/gc/vxsort/isa_detection.cpp

```diff
@@ -77,11 +77,11 @@ SupportedISA DetermineSupportedISA()
     // get xcr0 register
     DWORD64 xcr0 = _xgetbv(0);
 
-    // get OS XState info 
+    // get OS XState info
     DWORD64 FeatureMask = GetEnabledXStateFeaturesHelper();
 
     // get processor extended feature flag info
-    __cpuid(reg, 7);
+    __cpuidex(reg, 7, 0);
```

Comment:
@dotnet/gc. FYI that I'm fixing this here.

On Windows, `__cpuid` always explicitly zeros `ECX`. On Unix, it does not.

For some CPUID queries, the value of `ECX` is irrelevant. For others, such as `function_id: 0x07`, the value of `ECX` is very important and not explicitly passing down zero "can" result in a different set of ISA flags being returned than we want.

This was the only existing place that looked to be relying on the pre-existing value of ECX.

Reply:
This wasn't problematic so far since its under a `TARGET_WINDOWS` ifdef, but its still something that's good to ensure is "correct".

Reply:
assume this was found with just a `grep` and didnt cause any test failures?

Reply:
> assume this was found with just a grep and didnt cause any test failures?

Correct.

Reply:
we don't use this at all on linux. so this is not a must-have kind of thing but you could still do the change if you want. 

since we are on this topic though, and maybe you've done some research on this @tannergooding, could we just replace all this code with `__isa_available` when we compile with MSVC? that'd be much nicer than having to write this code ourselves.

Reply:
also CC-ing @PeterSolMS

Reply:
> could we just replace all this code with __isa_available when we compile with MSVC

There isn't really an equivalent to `__isa_available` today. But, could the GC maybe rely on the cached VM flags? That is, `JitFlags.GetInstructionSetFlags()` where `jitFlags` is passed down from the VM to `jitNativeCode`

Reply:
My mistake I was confusing `__isa_available` with `__builtin_cpu_supports` which is Clang/GCC only.

I don't see any docs on `__isa_available`, just a vague reference in an AVX512 related blog post indicating 6 or greater means AVX512F+BW+CD+DQ+VL

It seems like it could work for the GC but wouldn't be sufficient for the rest of the VM/JIT. It might be better for the VM to just pass the flags its already resolving on startup into the GC instead.